src/keystore.cpp

```diff
@@ -155,8 +155,20 @@ void CBasicKeyStore::AddSaplingFullViewingKey(
     auto ivk = fvk.in_viewing_key();
     mapSaplingFullViewingKeys[ivk] = fvk;
 
-    // Add defaultAddr -> SaplingIncomingViewing to SaplingIncomingViewingKeyMap
-    mapSaplingIncomingViewingKeys[defaultAddr] = ivk;
+    AddSaplingIncomingViewingKey(ivk, defaultAddr);
+}
+
+// This function updates the wallet's internal address->ivk map. 
+// If we add an address that is already in the map, the map will
+// remain unchanged as each address only has one ivk.
+void CBasicKeyStore::AddSaplingIncomingViewingKey(
```

Comment:
Make this return `bool`, since when we add wallet serialization this call may fail (at a higher layer).